src/GitVersion.Core/VersionCalculation/Mainline/MainlineContext.cs

```diff
@@ -4,11 +4,13 @@
 
 namespace GitVersion.VersionCalculation.Mainline;
 
-internal record MainlineContext(IIncrementStrategyFinder IncrementStrategyFinder, IGitVersionConfiguration Configuration)
+internal record MainlineContext(IIncrementStrategyFinder IncrementStrategyFinder, IGitVersionConfiguration Configuration, IGitRepository Repository, GitVersionContext GitverContext)
 {
     public IIncrementStrategyFinder IncrementStrategyFinder { get; } = IncrementStrategyFinder.NotNull();
 
     public IGitVersionConfiguration Configuration { get; } = Configuration.NotNull();
+    public IGitRepository Repository { get; } = Repository.NotNull();
```

Comment:
Not necessary in my opinion to extend the Context with IGitRepository and GitVersionContext.